pkg/routeagent/controllers/route/cleanuphandling.go

```diff
@@ -0,0 +1,18 @@
+package route
+
+import (
+	"github.com/submariner-io/submariner/pkg/routeagent/cleanuphandlers"
+	"k8s.io/klog"
+)
+
+func (r *Controller) InstallCleanupHandlers(handlers []cleanuphandlers.CleanupHandler) {
+	r.cleanupHandlers = append(r.cleanupHandlers, handlers...)
+}
+
+func (r *Controller) GatewayToNonGatewayTransition() {
+	for _, handler := range r.cleanupHandlers {
+		if err := handler.GatewayToNonGatewayTransition(); err != nil {
+			klog.Errorf("Error handling GatewayToNonGateway cleanup in %v, %s", handler, err)
```

Comment:
Is this where `GetName` was intended to be used?

```suggestion
			klog.Errorf("Error handling GatewayToNonGateway cleanup in %q, %s", handler.GetName(), err)
```